src/Transactions/CreateTransaction/CreateTransactionContainer.test.js

```diff
@@ -71,13 +85,36 @@ const renderCreateTransactionContainer = (props = {}) => render(
 );
 
 describe('CreateTransactionContainer', () => {
-  const selectFund = async (field = 'from', value = funds[0].code) => {
+  const selectFund = async (field, value) => {
+    // Select the specific button based on the field name attribute
+    const buttonSelector = screen.getByRole('button', { name: (_name, element) => element?.getAttribute('name')?.startsWith(field) });
+
+    await act(async () => user.click(buttonSelector));
+
+    // Wait for the dropdown options to appear and select the option
+    const fundOption = await screen.findByRole('option', { name: new RegExp(`.*${value}.*`, 'i') });
```

Comment:
is `findByRole` async method? maybe it's better to use it together with `waitFor`? 
and the same for other parts of code, I see that you use it together with `click` or `select`, but these functions are not async if I'm not mistaken

Reply:
https://testing-library.com/docs/dom-testing-library/api-async/#findby-queries
`findByRole<T extends HTMLElement = HTMLElement>(
        ...args: Parameters<BoundFunction<queries.FindByRole<T>>>
      ): ReturnType<queries.FindByRole<T>>`

https://testing-library.com/docs/user-event/convenience/#click
`click(element: Element): Promise<void>`